core/src/main/scala/filodb.core/store/ChunkSetInfo.scala

```diff
@@ -149,28 +149,30 @@ object ChunkSetInfo extends StrictLogging {
   def chunkSetInfoSize(numDataColumns: Int): Int = OffsetVectors + 8 * numDataColumns
   def blockMetaInfoSize(numDataColumns: Int): Int = chunkSetInfoSize(numDataColumns) + 4
 
-  def getChunkID(infoPointer: NativePointer): ChunkID = UnsafeUtils.getLong(infoPointer + OffsetChunkID)
+  def getChunkID(acc: MemoryReader, infoPointer: NativePointer): ChunkID = acc.getLong(infoPointer + OffsetChunkID)
   def getChunkID(infoBytes: Array[Byte]): ChunkID =
     UnsafeUtils.getLong(infoBytes, UnsafeUtils.arayOffset + OffsetChunkID)
   def setChunkID(infoPointer: NativePointer, newId: ChunkID): Unit =
     UnsafeUtils.setLong(infoPointer + OffsetChunkID, newId)
 
-  def getIngestionTime(infoPointer: NativePointer): Long = UnsafeUtils.getLong(infoPointer + OffsetIngestionTime)
+  def getIngestionTime(acc: MemoryReader, infoPointer: NativePointer): Long =
+    acc.getLong(infoPointer + OffsetIngestionTime)
   def getIngestionTime(infoBytes: Array[Byte]): Long =
     UnsafeUtils.getLong(infoBytes, UnsafeUtils.arayOffset + OffsetIngestionTime)
```

Comment:
1 & 2.  The presence of accessor and NativePointer is a source of confusion. I changed the signature to use `infoAcc: MemoryReader, infoAddr: Long` as params. This should make it clear and it is also applicable to Native Pointers.

3. Some ODP workflows are still using the byte-array APIs. Since this PR is already large, I dont want to touch that code and risk regressions. I will mark as deprecated in lieu of other API.